src/plugins/robots/drone/hardware/drone_flight_system_default_sensor.cpp

```diff
@@ -104,6 +119,22 @@ namespace argos {
          /* clear out the read data */
          m_tAttitude.reset();
       }
+      if(m_tAttitudeTarget) {
+         const mavlink_attitude_target_t& tReading =
+            m_tAttitudeTarget.value();
+         CQuaternion cTargetOrientation;
```

Comment:
Use the value constructor, i.e.: `CQuaternion cTargetOrientation(tReading.q[0], tReading.q[1], tReading.q[2], tReading.q[3]);`